evalml/automl/automl_search.py

```diff
@@ -627,6 +636,27 @@ def __init__(
             custom_hyperparameters=custom_hyperparameters,
         )
 
+    def _catch_warnings(self, warning_list):
```

Comment:
Allows us to catch all warnings thrown during pipeline initialization and only throw the warnings for the component names that appear in all warnings. For instance, if we had pipelines:
```python
p1 = ['Imputer', 'Standard Scaler', 'Logistic Regression Classifier']
p2 = ['Imputer', 'Random Forest Classifier']
params_passed_in = {"Logistic Regression Classifier": {"some value": 2}, 
                   {"Undersampler": {"another val": 1}}
```
then we would only want to raise warning for `Undersampler` since it isn't used in any of the pipelines

Reply:
Tbh when I first read the issue I didn't foresee we would need this in AutoMLSearch but I think it's the right call!